src/sandbox/pinned.rs

```diff
@@ -1,8 +1,69 @@
-//! A pod sandbox implementation which does pin it's namespaces to file descriptors.
+//! A pod sandbox implementation which does pin its namespaces to file descriptors.
 
-use crate::sandbox::Pod;
+use crate::sandbox::{LinuxNamespaces, Pod, SandboxData};
+use async_trait::async_trait;
+use anyhow::{bail, Context, Result};
+use tokio::process::Command;
+use which::which;
+use std::path::PathBuf;
 
 #[derive(Default)]
-pub struct PinnedSandbox {}
+pub struct PinnedSandbox {
+    pinner_path: PathBuf,
+    namespaces_dir: String,
+    ready: bool,
+}
 
-impl Pod for PinnedSandbox {}
+impl PinnedSandbox {
+    pub fn new(pinner: &str) -> Result<PinnedSandbox> {
```

Comment:
@saschagrunert We want to be able to setup a PinnedSandbox::new and then use it in the Sandbox but that doesn't work  so we may need to use trait objects.

Reply:
Hm, yes it might be the better approach to use trait objects here. :+1: